drs_cli/client.py

```diff
@@ -30,18 +33,27 @@ class DRSClient():
     def __init__(
         self,
         host: str = 'http://0.0.0.0',
-        port: int = '8080',
+        port: int = 80,
         base_path: str = 'ga4gh/drs/v1',
+        drs_uri: Optional[str] = None,
         token: Optional[str] = None,
-    ) -> None:
+    ) -> Union[None, Error]:
+        
         self.url = f"{host}:{port}/{base_path}"
+        if drs_uri:
+            if _check_drs_regex(drs_uri):
+                drs_host = _get_host_and_id(drs_uri)[0]
+                self.url = f"http://{drs_host}/{base_path}"
+            else:
+                return Error(**DRS_URI_FORMAT_ERROR)
         self.token = token
         self._get_headers()
         logger.info(f"API URL: {self.url}")
 
     def get_object(
         self,
-        object_id: str,
+        object_id: Optional[str] = None,
+        drs_uri: Optional[str] = None,
```

Comment:
Same thing as before: just use `object` which can be either an object ID or a DRS URI, for the user's convenience. Then do the regex checks (one for DRS URI and one for allowed object IDs, as defined in the DRS specs) to determine which was passed

Reply:
Should I leave this as it was before, coz I have added regex for HTTP, HTTPS URIs, and also regex for DRS IDs. It improves the readability of the code.